c2corg_ui/static/js/documentservice.js

```diff
@@ -78,62 +104,71 @@ app.Document.prototype.hasAssociation = function(type, id) {
 
 /**
  * @param {appx.SimpleSearchDocument} doc
+ * @param {string=} doctype Optional doctype
+ * @param {boolean=} setOutingTitle
  * @export
  */
-app.Document.prototype.pushToAssociations = function(doc) {
-  var routesLength =  this.document.associations['routes'].length;
+app.Document.prototype.pushToAssociations = function(doc, doctype,
+    setOutingTitle) {
   var associations = this.document.associations;
-  var doctype = app.utils.getDoctype(doc['type']);
+  doctype = doctype || app.utils.getDoctype(doc['type']);
+  setOutingTitle = typeof setOutingTitle !== 'undefined' ?
+    setOutingTitle : false;
 
-  if (doctype === 'routes') {
-    // if there are no routes = new document => set title to the title of the first associated route
-    if (!routesLength) {
-      this.document['locales'][0]['title'] = doc['locales'][0]['title'];
-    }
-  }
   associations[doctype].push(doc);
-  return;
+
+  // When creating an outing, the outing title defaults to the title
+  // of the first associated route.
+  if (setOutingTitle && doctype === 'routes' &&
+      !this.document.locales[0]['title'] &&
+      this.document.associations.routes.length == 1) {
+    var title = 'title_prefix' in doc.locales[0] &&
+      doc.locales[0]['title_prefix'] ?
+      doc.locales[0]['title_prefix'] + ' : ' : '';
+    title += doc.locales[0]['title'];
+    this.document.locales[0]['title'] = title;
```

Comment:
This is the default title when creating an outing, based upon the title of the first associated route. If the title is already set, this block would not be run at all thanks to the condition ``!this.document.locales[0]['title']``.